### DISCUSSION TITLE
Feature Request: ability to use prepared statement parameters with interval syntax

### DISCUSSION THREAD

**Discussion by dylanscott** [duckdb/duckdb#6935] [Ideas]
I don't feel at all strongly about this because there's an easy workaround, but just filing in case it's easy.

DuckDB supports this [nice interval syntax](https://duckdb.org/docs/sql/functions/interval) where you can say `interval 1 hour` rather than quoting like `interval '1 hour'` (which I think you always have to quote in Postgres?). But then it's tempting to parameterize this syntax like

```python
import duckdb

conn = duckdb.connect(database=':memory:')

# using prepared statement param with interval syntax
print(conn.execute("""
select current_timestamp - interval ? hour
""", [1]).fetch_df())
```

But this gives a parse error

```
duckdb.ParserException: Parser Error: syntax error at or near "?"
LINE 2: select current_timestamp - interval ? hour
```

Using the corresponding interval function works though, so there's an easy workaround

```python
# using prepared statement param with interval fn
print(conn.execute("""
select current_timestamp - to_hours(?)
""", [1]).fetch_df())
```

**Comment by Mytherin** [comment]
You can do this currently - but you need to use brackets, e.g.:

```sql
prepare v1 as select current_timestamp - interval (?) hours as ts;
execute v1(42);
┌───────────────────────────┐
│            ts             │
│ timestamp with time zone  │
├───────────────────────────┤
│ 2023-03-30 02:35:48.14+02 │
└───────────────────────────┘
```

I think this triggers an ambiguity in the parser otherwise, but I can have a look to see if this is fixable.

**Reply by Mytherin** [reply]
Yeah, unfortunately the current parser does not like the variant without the brackets.
